src/processor/reconnection-graph.service.ts

```diff
@@ -125,4 +136,179 @@ export class ReconnectionGraphService implements OnApplicationBootstrap, OnAppli
+    graphConnections.forEach((connection) => {
+      let schemaId: number;
+      const connectionType = connection.connectionType.toLowerCase();
+      const privacyType = connection.privacyType.toLowerCase();
+
+      switch (connectionType) {
+        case 'follow':
+          schemaId = privacyType === 'public' ? public_follow_schema_id : private_follow_schema_id;
+          break;
+        case 'friendship':
+          schemaId = privacyType === 'public' ? public_friendship_schema_id : private_friendship_schema_id;
+          break;
+        default:
+          throw new Error(`Unrecognized connection type: ${connectionType}`);
+      }
+
+      switch(connection.direction) {
+        case 'connectionTo': {
+          let connectionAction = {
+            type: "Connect",
+            ownerDsnpUserId: dsnpUserId.toString(),
+            connection: {
+              dsnpUserId: connection.dsnpId,
+              schemaId,
+            } as Connection,
+          } as ConnectAction;
+
+          if (dsnpKeys) {
+            connectionAction['dsnpKeys'] = dsnpKeys;
+          }
+          actions.push(connectionAction);
+          break;
+        }
+        case 'connectionFrom':{
+          const { key: jobId, data } = createGraphUpdateJob(connection.dsnpId, providerId, SkipTransitiveGraphs);
+          this.graphUpdateQueue.add('graphUpdate', data, { jobId });
+          break;
+        }
+        case 'bidirectional':{
+          let connectionAction = {
+            type: "Connect",
+            ownerDsnpUserId: dsnpUserId.toString(),
+            connection: {
+              dsnpUserId: connection.dsnpId,
+              schemaId,
+            } as Connection,
+          } as ConnectAction;
+
+          if (dsnpKeys) {
+            connectionAction['dsnpKeys'] = dsnpKeys;
+          }
+          actions.push(connectionAction);
+          const { key: jobId, data } = createGraphUpdateJob(connection.dsnpId, providerId, SkipTransitiveGraphs);
+          this.graphUpdateQueue.add('graphUpdate', data, { jobId });
+          break;
+        }
+        default:
+          throw new Error(`Unrecognized connection direction: ${connection.direction}`);
+      }
+    });
+    return actions;
+  }
+
+  async formDsnpKeys(dsnpUserId: MessageSourceId, graphSdkConfig: Config): Promise<DsnpKeys | undefined> {
+    const public_key_schema_id = graphSdkConfig.graphPublicKeySchemaId;
+    let publicKeys: ItemizedStoragePageResponse = await this.api.rpc.statefulStorage.getItemizedStorage(dsnpUserId, public_key_schema_id);
+
+    let dsnpKeys: DsnpKeys | undefined;
+    if (publicKeys.items.length > 0) {
+      dsnpKeys = {
+        dsnpUserId: dsnpUserId.toString(),
+        keysHash: publicKeys.content_hash.toNumber(),
+        keys: publicKeys.items.map((item: ItemizedStorageResponse) => {
+          return {
+            index: item.index.toNumber(),
+            content: item.payload.toU8a(),
+          } as KeyData;
+        }),
```

Comment:
Use declared return type instead of casting to get strong type checking

```suggestion
        keys: publicKeys.items.map((item: ItemizedStorageResponse): KeyData => ({
            index: item.index.toNumber(),
            content: item.payload.toU8a(),
          })),
```